clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -600,6 +602,30 @@ public void testCoordinatorNotAvailable() {
         assertTrue(coordinator.coordinatorUnknown());
     }
 
+    @Test
+    public void testSyncCommitWithoutOffsetsAndPendingAsyncCommit() {
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+
+        TopicPartition tp = new TopicPartition("foo", 0);
+        Map<TopicPartition, OffsetAndMetadata> offsets = singletonMap(tp, new OffsetAndMetadata(123));
+
+        final AtomicBoolean committed = new AtomicBoolean();
```

Comment:
does this need to be atomic? there shouldn't be concurrent access to the var.

Reply:
Probably because of this:
<img width="539" alt="afbeelding" src="https://github.com/apache/kafka/assets/630600/847654a4-8b22-49c8-9cb0-534b7586e4be">